src/main/java/com/commercetools/sync/services/impl/ProductServiceImpl.java

```diff
@@ -47,15 +50,26 @@ public ProductServiceImpl(@Nonnull final ProductSyncOptions syncOptions) {
             productKeys,
             keysNotCached -> ProductQuery
                 .of()
-                .withPredicates(buildResourceKeysQueryPredicate(keysNotCached)));
+                .withPredicates(buildProductKeysQueryPredicate(keysNotCached)));
+    }
+
+    QueryPredicate<Product> buildProductKeysQueryPredicate(@Nonnull final Set<String> productKeys) {
+        final List<String> keysSurroundedWithDoubleQuotes = productKeys.stream()
+                                                                       .filter(StringUtils::isNotBlank)
+                                                                       .map(productKey -> format("\"%s\"", productKey))
+                                                                       .collect(Collectors.toList());
```

Comment:
Can you change this to `Collectors.joining(",")` and then you can remove the lines:
```
        String keysQueryString = keysSurroundedWithDoubleQuotes.toString();
        // Strip square brackets from list string. For example: ["key1", "key2"] -> "key1", "key2"
        keysQueryString = keysQueryString.substring(1, keysQueryString.length() - 1);
```